File: source/frontend/src/constants/financialCodeTypes.ts
export enum FinancialCodeTypes {
  BusinessFunction = 'BusinessFunction',
  CostTypes = 'CostTypes',
  WorkActivity = 'WorkActivity',
  ChartOfAccounts = 'ChartOfAccounts',
  FinancialActivity = 'FinancialActivity',
  Responsibility = 'Responsibility',
  YearlyFinancial = 'YearlyFinancial',
}

export function formatFinancialCodeType(value: FinancialCodeTypes): string {
  switch (value) {
    case FinancialCodeTypes.BusinessFunction:
      return 'Business function';
    case FinancialCodeTypes.CostTypes:
      return 'Cost types';
    case FinancialCodeTypes.WorkActivity:
      return 'Work activity';
    case FinancialCodeTypes.ChartOfAccounts:
      return 'Chart of accounts';
    case FinancialCodeTypes.FinancialActivity:
      return 'Financial activity';
    case FinancialCodeTypes.Responsibility:
      return 'Responsibility';
    case FinancialCodeTypes.YearlyFinancial:
      return 'Yearly financial';
    default:
      return 'Unknown';
  }
}

File: source/backend/api/Models/Concepts/FinancialCode/FinancialCodeTypes.cs
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace Pims.Api.Models.Concepts
{
    /// <summary>
    /// Available financial code lookups.
    /// </summary>
    [JsonConverter(typeof(JsonStringEnumMemberConverter))]
    public enum FinancialCodeTypes
    {
        [EnumMember(Value = "business-function")]
        BusinessFunction,

        [EnumMember(Value = "cost-types")]
        CostType,

        [EnumMember(Value = "work-activity")]
        WorkActivity,

        [EnumMember(Value = "chart-of-accounts")]
        ChartOfAccounts,

        [EnumMember(Value = "financial-activity")]
        FinancialActivity,

        [EnumMember(Value = "responsibility")]
        Responsibility,

        [EnumMember(Value = "yearly-financial")]
        YearlyFinancial,
    }
}
